src/project.cc

```diff
@@ -121,6 +121,12 @@ Project::Entry GetCompilationEntryFromCompileCommandEntry(
   // Include the compiler in the args.
   if (i > 0)
     result.args.push_back(entry.args[i - 1]);
+  else {
+    // Args probably came from a /.cquery file, which likely has just flags.
+    // clang_parseTranslationUnit2FullArgv() expects the binary path as the
+    // first arg, so the first flag would end up being ignored. Add a dummy.
+    result.args.push_back("clang++");
```

Comment:
`README.md` should be changed to require specifying compiler driver (gcc/clang++/...)

Reply:
from the clang docs, the only benefit of specifying this (since we force the `-x` value from the file type) is to help find the standard library if it's relative to the binary path.  so, that's what I added to the docs; let me know if it should say something else.  I do think it'd be a common mistake to not include it (even if the docs tell you to), so I think the code workaround is still beneficial.